src/api/app/controllers/webui/kiwi/images_controller.rb

```diff
@@ -20,7 +20,8 @@ def import_from_package
         if package.kiwi_image.blank? || package.kiwi_image.destroyed?
           package.kiwi_image = ::Kiwi::Image.build_from_xml(package.source_file(kiwi_file), package.kiwi_file_md5)
           unless package.save
-            errors = ["Kiwi File '#{kiwi_file}' has errors:", package.kiwi_image.errors.full_messages].join('<br />')
+            errors = image_error_messages("Kiwi File '#{kiwi_file}' has errors:",
+                                          package.kiwi_image, [package.kiwi_image.package_groups.map(&:packages)].flatten.compact)
             redirect_to package_view_file_path(project: package.project, package: package, filename: kiwi_file), error: errors
```

Comment:
Why render the show page like its done in #update instead of redirecting?
```ruby
render action: :show
```

Reply:
Because we are redirecting to package_view_file and this needs to set other instances variables.